### PR TITLE
Computation of boundaries using zscores

### PR DESCRIPTION
**Pull request by fidelram**
This new method uses zscores to compute TAD-separation scores and call boundaries. The advantages of using zscores is the reduction of noise and the normalization of the data.

### CODE REVIEW

**Comment by fidelram** [comment]
@bgruening this new version creates several new files. I don't know how to handle that with galaxy. The number of output files varies depending on one of the parameters `--delta`. If multiple `--delta` values are given, then the output contains 2* number of values. If this can not be handled by Galaxy, a workaround is to pass only one `--delta` parameter. 

I documented the files created in the help section.

File: scripts/hicComputeSaturation.py [hunk 1]
```diff
@@ -1,159 +0,0 @@
-#!/usr/bin/env python
-#-*- coding: utf-8 -*-
-from __future__ import division
```
Comment (bgruening) [inline scripts/hicComputeSaturation.py]: Is this intended?

File: hicexplorer/hicFindTADs.py [hunk 2]
```diff
@@ -764,7 +623,22 @@ def get_domains(boundary_list):
     return domain_list
 
 
-def save_bedgraph_matrix(outfile, chrom, chr_start, chr_end, score_matrix):
+def toBytes(s):
```
Comment (bgruening) [inline hicexplorer/hicFindTADs.py]: @fidelram maybe you want to use six here? http://www.programcreek.com/python/example/54051/six.BytesIO

File: hicexplorer/hicFindTADs.py [hunk 3]
```diff
@@ -900,13 +781,18 @@ def compute_spectra_matrix(args, matrix=None):
...
+
     # remove self counts
     hic_ma.diagflat(value=0)
 
     # mask bins without any information
     hic_ma.maskBins(hic_ma.nan_bins)
     orig_intervals = hic_ma.cut_intervals
 
+    # use zscore matrix
+    sys.stderr.write("Computing z-score matrix...\n")
```
Comment (bgruening) [inline hicexplorer/hicFindTADs.py]: Can we write to stdout?

File: hicexplorer/hicFindTADs.py [hunk 4]
```diff
@@ -932,6 +818,7 @@ def compute_spectra_matrix(args, matrix=None):
 
     binsize = hic_ma.getBinSize()
 
+    sys.stderr.write("Computing TAD-separation scores...\n")
```
Comment (bgruening) [inline hicexplorer/hicFindTADs.py]: Same as above.

**Comment by bgruening** [comment]
@fidelram this is possible. An example is here: https://github.com/galaxyproject/galaxy/blob/71cea6604d43c5fe6215f5656462ba6c1af69bb6/test/functional/tools/multi_output_configured.xml

However, I'm unclear about the usability about this. Feel free to merge and I will integrate the multiple outputs, post-merge. Thanks!

File: hicexplorer/hicFindTADs.py (hunk 3 above)
Comment (fidelram) [inline hicexplorer/hicFindTADs.py]: This is not standardized in the code but in general, I am trying to output messages using standard error. Standard ouput is reserved for output that may be passed to other programs.

File: scripts/hicComputeSaturation.py (hunk 1 above)
Comment (fidelram) [inline scripts/hicComputeSaturation.py]: Actually yes, I want 1/2 = 0.5.

**Comment by fidelram** [comment]
@bgruening I reduced the number of files that are output which are four:

 * A bed file with the boundaries
 * A bed file with the domains
 * A bedgraph file with the TAD-separation score
 * A GFF file with the boundaries and extra data.

The files are named as `<prefix>_delta0.01_pvalue0.01_boundaries.bed`, `<prefix>_delta0.01_pvalue0.01_domains.bed`, `<prefix>_delta0.01_pvalue0.01_boundaries.gff` and `<prefix>_score.bedgraph`

**Pull request merged**